### DISCUSSION TITLE
Porting bz to unix/macOS

### DISCUSSION THREAD

**Discussion by kamischi** [openmopac/mopac#159] [General]
So far bz is using windows routines from ifort. How about porting bz to unix/macOS? 

Which GUI library should be used? Qt, GTK, ...

I found fortran-sdl2 (https://github.com/interkosmos/fortran-sdl2). Anyone?

Anyone interested as programmer or user?

Greetings - Michael.

**Comment by godotalgorithm** [comment]
Right, bz and winmopac use the QuickWin library to display text or graphics in window. QuickWin is an ancient library that is still supported by Intel, but it has a lot of technical limitations and I wouldn't be surprised if it eventually gets abandoned. I have no plans to continue supporting its use in MOPAC.

I would welcome anyone's attempt to port either of these features to any cross-platform alternative. MOPAC is using Qt right now to build graphical installers, so that could be a natural choice. I have no experience in GUI development other than building the graphical installers for MOPAC, but that was done using Javascript, not Fortran.

I don't think bz is used very much, so it is a low development priority for me right now.

**Comment by godotalgorithm** [comment]
While I need to explore this a bit more, my current understanding is that the graphical components of BZ are not especially interactive. Rather than display graphics in a window, it might make more sense to adapt this program to prepare graphics as a PDF file. Fortran doesn't seem to have convenient libraries for plotting figures to PDF, but Python makes this very convenient and portable with matplotlib. It will take some effort to decide how best to approach this adaptation - the BZ program is doing a lot of preprocessing before plotting anything, and it would be quite a lot of work to port all of that preprocessing to Python. Another approach might be to retain the Fortran program and have it output its data in a format that is convenient for a follow-on Python script that generates the PDF plots from the data. Alternatively, the program could be refactored into a library that is called by a Python script before using matplotlib to render the output data.
